File: tests/expectations/simplify-option-ptr.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

typedef struct Opaque Opaque;

typedef struct {
  const Opaque *x;
  Opaque *y;
  void (*z)(void);
} Foo;

typedef union {
  const Opaque *x;
  Opaque *y;
  void (*z)(void);
} Bar;

void root(const Opaque *a, Opaque *b, Foo c, Bar d);

File: tests/expectations/simplify_option_ptr.zig
const std = @import("std");

pub const Opaque = opaque {};
pub const Option_____Opaque = opaque {};

pub const Foo = extern struct {
    x: ?*const Opaque,
    y: ?*Opaque,
    z: ?*const fn () anyopaque,
    zz: [*]?*const fn () anyopaque,
};

pub const Bar = extern union {
    x: ?*const Opaque,
    y: ?*Opaque,
    z: ?*const fn () anyopaque,
    zz: [*]?*const fn () anyopaque,
};

pub extern fn root(a: ?*const Opaque, b: ?*Opaque, c: Foo, d: Bar, e: ?*Option_____Opaque, f: ?*const fn (?*const Opaque) anyopaque) anyopaque;
